### PR TITLE
Add caching to config invocation handler

### PR DESCRIPTION
**Pull request by KGoodale13**
I added caching for resolving the annotation values of config classes in the config invocation handler and also added caching for getting the value of a config entry to minimize the amount of stringToObject and callDefaultMethod calls.

### CODE REVIEW

**Comment by deathbeam** [comment]
Travis is still failing

File: runelite-client/src/main/java/net/runelite/client/config/ConfigManager.java [hunk 1]
```diff
@@ -267,6 +284,10 @@ public String getConfiguration(String groupName, String key)
 	{
 		return properties.getProperty(groupName + "." + key);
 	}
+	public String getConfiguration(String propertyKey)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/config/ConfigManager.java]: Empty line above

**Comment by KGoodale13** [comment]
Styling is finally fixed

**Comment by raiyni** [comment]
Added in https://github.com/runelite/runelite/commit/c342f7391a7e56330d156b3db44085fbe68453a5

**Pull request closed without merging**